rdrf/registry/patients/models.py

```diff
@@ -756,19 +758,38 @@ def hard_delete_clinicaldata(self, patient_id):
         for cd in clinicaldata_models:
             cd.delete()
 
-    def set_consent(self, consent_model, answer=True, commit=True):
+    def notify_consent_change(self, consent_value):
+        if self.clinician and not consent_value.answer:
+            logger.info(f"Notify clinician {self.clinician} for change on {consent_value.consent_question}")
+            template_data = {
+                'patient': self,
+                'clinician': self.clinician,
+                'consent_question': consent_value.consent_question,
+                'answer': consent_value.answer,
+            }
+            registry = self.rdrf_registry.first()
+            process_notification(
+                registry.code,
+                EventType.PATIENT_CONSENT_CHANGE,
+                template_data
+            )
+
+    def set_consent(self, consent_model, answer=True, commit=True, notify_clinician=False):
         patient_registries = [r for r in self.rdrf_registry.all()]
         if consent_model.section.registry not in patient_registries:
             return   # error?
         cv, created = ConsentValue.objects.get_or_create(
             consent_question=consent_model, patient=self)
+        answer_changed = not created and cv.answer != answer
         cv.answer = answer
         if cv.first_save:
             cv.last_update = datetime.datetime.now()
         else:
             cv.first_save = datetime.datetime.now()
         if commit:
             cv.save()
+            if notify_clinician and answer_changed:
```

Comment:
As discussed, drop the `notify_clinician` and don't send 1 email notification on each value change.
We want to send the notification once when the consent form was saved if any of the consent question values changed.

Reply:
Updated.